### ISSUE TITLE
TUID and privacy

### ISSUE THREAD

**Issue by petrkr** [covid19cz/erouska-android#179]
> Transmitted user ID, actually transmitted over Bluetooth, each BUID gets about 100 TUIDs after registration, client picks a different one every hour or so, privacy measure to prevent tracking individual device/user, 10 bytes

What happen when all 100 TUID's will be exhausted? Start reusing old ones or will somehow ask server for new range. Or re-register?

You call about privacy first. So why you collect data about user's phone such as Mode/Manufacture ? That could be really good big data and could be sold to some statistics company for who those data are important in marketing.

**Comment by v60coffee** [comment]
They do not care about privacy so much. See their design and stack choices. Could have gone with notifications-after-infection-possibility. And then one could provide a mean of contact, be it a state line phone, email or mobile phone. But no, the want to hold your mobile number hostage from the beginning.

**Comment by davidvavra** [comment]
When all 100 TUIDs are exhausted, we start rotating them. We believe it's enough for preventing people tracking others via stable id. We can start generating more than 100 on the backend, if it's needed.

Manufacturer/model is needed for a case where you register multiple phones to a single phone number. People from hygiene see that number +420123456 (Galaxy S8) was dangerously close to the infected.

But this is more a question/discussion than an issue. GitHub issues is a place for actual bugs in the app, not for discussion. Please contact our support at info@erouska.cz with further questions.

**Comment by jendakol** [comment]
@petrkr 
Ad TUIDs: TUIDs are reused. Do you see a big problem with that? I mean, it's like using each ID once per week... Could you really imagine someone who uses this to attack?

Ad phone data: I don't know if these data could be sold but that's definitely not what we're doing. Let me add something to the response of David.   
**TLDR**: These data are collected because we need to adjust measured values on the backend.  
**Full version**: The problem is that each BT chip has different broadcasting power and in order to be as precise as possible (because at the end of the day we need _distance_, not _signal strength_) we need to do adjustments. We use the manufacturer/model to determine how the signal has to be adjusted.

@v60coffee Thank you for not responding to stuff you don't understand. There are reasons why we've chosen this solution among many others. Believe me and save your hates, please. The reason is explained on our site as well as like million times at Facebook etc. (where discussions like this one belong).

Please don't misuse issues in the repository to discussions.

**Comment by petrkr** [comment]
> Please don't misuse issues in the repository to discussions

issues are commonly on github used to discuss over code. What was my question. Discuss about what is "right" about law or privacy I agree it not right place here.

Anyway back to code:
you wrote you rotate them, but I meanwhile found this:

```
fun getRandomTuid(lastTuid: String? = null) : String?{
        val stringSet = prefs.getStringSet(DEVICE_TUIDS, emptySet())
        return stringSet?.let {
            if (it.size > 1) { // make sure we don't loop in here forever
                it.random()?.run {
                    if (this == lastTuid) getRandomTuid(lastTuid) else this
                }
            } else {
                it.firstOrNull()
            }
        }
    }
```

which means it can re-use it more offten than once-per 100 usages.. Maybe consider do something like "db of used uuids" and after unused are exhausted, let's rotate whole again ?

Atleast it would delay reuse really to that 1 week (100, 1 per hour == 4.1 days if app will run 24/7)

Maybe good idea also would be generate 1000 even it would took 10x more data storage (which is still nothing compare to todays size) and would increase privacy to aprox 41 days.

**Comment by jendakol** [comment]
@petrkr Hi, the "don't misuse issues in the repository to discussions" was not pointed to you, sorry for the confusion.

We're currently discussing how to rotate TUIDs and even their amount. We've used the random strategy so far because it's not that predictable.

Thanks for input. Maybe it would be better to try next time without accusing us from selling data :smiley:
